### PR TITLE
App Service cmd for importing certs from Key Vault

### PR DESCRIPTION
**Pull request by madsd**
App Service cmd for importing certs from Key Vault:
* az webapp|functionapp config ssl import

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by haroldrandom** [comment]
Please run `azdev linter {YOUR_MODULE}` to make sure command's help message is valid

**Comment by madsd** [comment]
Hi @haroldrandom. I fixed the help error and some other styling. I also added a scenario test. The test works when I run it local, but fails in CI - the error is from creating an app service plan. The same lines of code found in numerous other tests. Any idea what is wrong?

**Comment by haroldrandom** [comment]
@madsd Have you edited the recording YAML file manually?

**Comment by yonzhan** [comment]
@jiasli  please help with the PR.

**Comment by madsd** [comment]
No changes to the Yaml file, and I can run the tests in live mode on my machine without problems. The strange thing is, that it seem to fail on the basic appservice plan create. Don't know if there is a way to get more details on the exception.

**Comment by panchagnula** [comment]
> No changes to the Yaml file, and I can run the tests in live mode on my machine without problems. The strange thing is, that it seem to fail on the basic appservice plan create. Don't know if there is a way to get more details on the exception.

you can delete the YAML file locally & re-record & upload the new yaml. You can also do is do a azdevsetup -c again on your local machine & see if you see this failure locally as well.

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py [hunk 1]
```diff
@@ -143,6 +143,8 @@ def load_command_table(self, _):
         g.custom_command('bind', 'bind_ssl_cert', exception_handler=ex_handler_factory(), validator=validate_app_or_slot_exists_in_rg)
         g.custom_command('unbind', 'unbind_ssl_cert', validator=validate_app_or_slot_exists_in_rg)
         g.custom_command('delete', 'delete_ssl_cert', exception_handler=ex_handler_factory())
+        g.custom_command('import', 'import_ssl_cert', exception_handler=ex_handler_factory(), is_preview=True)
+        g.custom_command('create', 'create_ssl_cert', exception_handler=ex_handler_factory(), is_preview=True)
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/commands.py]: Give we have AppService Certificates & Managed Certificates  this is targetting Free/Mamanged Certs only we should update this to name to reflect that this is for managed certificate by saying create-managed-certificate or something like that?

Comment (madsd) [reply src/azure-cli/azure/cli/command_modules/appservice/commands.py]: Managed Certificate removed for now.

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -2005,6 +2005,52 @@ def delete_ssl_cert(cmd, resource_group_name, certificate_thumbprint):
...
+
+
+def _check_service_principal_permissions(cmd, resource_group_name, key_vault_name):
+    from azure.cli.command_modules.keyvault._client_factory import keyvault_client_vaults_factory
+    from azure.cli.command_modules.role._client_factory import _graph_client_factory
+    from azure.graphrbac.models import GraphErrorException
+    kv_client = keyvault_client_vaults_factory(cmd.cli_ctx, None)
+    vault = kv_client.get(resource_group_name=resource_group_name, vault_name=key_vault_name)
+    # Check for Microsoft.Azure.WebSites app registration (app-id: abfa0a7c-a6b6-4736-8310-5855508787cd)
+    AZURE_PUBLIC_WEBSITES_APP_ID = 'abfa0a7c-a6b6-4736-8310-5855508787cd'
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py]: where to get these app Ids? is  it possible to import frorm dependencies?

**Comment by yungezz** [comment]
is it necessary to support version of cert?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 2 above)
Comment (madsd) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: No, these are "well-known" guids and are documented in this official blog: https://azure.github.io/AppService/2016/05/24/Deploying-Azure-Web-App-Certificate-through-Key-Vault.html

**Comment by madsd** [comment]
> is it necessary to support version of cert?

No, the commands setups up the integration with KV and pulls the latest version - any future versions of the certs are handled directly between App Service and Key Vault.

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 3]
```diff
@@ -2005,6 +2005,52 @@ def delete_ssl_cert(cmd, resource_group_name, certificate_thumbprint):
...
+    location = webapp.location
+    kv_id = _validate_key_vault_id(cmd.cli_ctx, key_vault, resource_group_name)
+    kv_id_parts = parse_resource_id(kv_id)
+    kv_name = kv_id_parts['name']
+    kv_resource_group_name = kv_id_parts['resource_group']
+    cert_name = '{}-{}-{}'.format(resource_group_name, kv_name, key_vault_certificate_name)
+
+    if not _check_service_principal_permissions(cmd, kv_resource_group_name, kv_name):
+        logger.warning('Unable to verify Key Vault permissions.')
+        logger.warning('You may need to grant Microsoft.Azure.WebSites service principal the Secret:Get permission')
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py]: Add link to the AppService blog here https://azure.github.io/AppService/2016/05/24/Deploying-Azure-Web-App-Certificate-through-Key-Vault.html

Comment (madsd) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: Added link to warning (and renamed validate function to format per discussion)

**Comment by madsd** [comment]
@yungezz, @haroldrandom requested changes has been implemented and the the latest History.rst conflict resolved.

**Pull request merged**